pkg/controller/istiocontrolplane/inputs.go

```diff
@@ -0,0 +1,91 @@
+package istiocontrolplane
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/helm/pkg/manifest"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha1"
+	"istio.io/operator/pkg/helmreconciler"
+)
+
+// defaultProcessingOrder for the rendered charts
+var defaultProcessingOrder = []string{
+	"istio",
+	"istio/charts/security",
+	"istio/charts/galley",
+	"istio/charts/prometheus",
+	"istio/charts/mixer",
+	"istio/charts/pilot",
+	"istio/charts/gateways",
+	"istio/charts/sidecarInjectorWebhook",
+	"istio/charts/grafana",
+	"istio/charts/tracing",
+	"istio/charts/kiali",
+}
+
+// IstioInputFactory creates a RenderingInput specific to an IstioControlPlane instance.
+type IstioInputFactory struct{}
+
+var _ helmreconciler.RenderingInputFactory = &IstioInputFactory{}
+
+// IstioRenderingInput is a RenderingInput specific to an IstioControlPlane instance.
+type IstioRenderingInput struct {
+	instance *v1alpha1.IstioControlPlane
+}
+
+var _ helmreconciler.RenderingInput = &IstioRenderingInput{}
+
+// NewRenderingInput creates a new IstioRenderiongInput for the specified instance.
+func (f *IstioInputFactory) NewRenderingInput(instance runtime.Object) (helmreconciler.RenderingInput, error) {
+	istioControlPlane, ok := instance.(*v1alpha1.IstioControlPlane)
```

Comment:
NewFoo functions should generally avoid errors if possible. You can avoid it by taking IstioControlPlane as input here.